mods/classic-taskbar-buttons-lite-vs-without-spacing.wh.cpp

```diff
@@ -0,0 +1,139 @@
+*/
+// ==/WindhawkModReadme==
+
+#include <windhawk_utils.h>
+#include <windows.h>
+#include <uxtheme.h>
+#include <vssym32.h>
+
+typedef struct tagBUTTONRENDERINFOSTATES {
+    char data[12];
+} BUTTONRENDERINFOSTATES, *PBUTTONRENDERINFOSTATES;
+
+/* Remove clock hover effect */
+typedef void (*__cdecl GDIHelpers_FillRectARGB_t)(HDC, LPCRECT, BYTE, DWORD, bool);
+GDIHelpers_FillRectARGB_t GDIHelpers_FillRectARGB_orig;
+void __cdecl GDIHelpers_FillRectARGB_hook(
+    HDC     hDC,
+    LPCRECT lprc,
+    BYTE    btUnknown,
+    DWORD   dwUnknown,
+    bool    bUnknown
+)
+{
+    return;
+}
+
+/* Draw taskbar item */
+typedef void (*__cdecl CTaskBtnGroup__DrawBar_t)(void *, HDC, void *, void *);
+CTaskBtnGroup__DrawBar_t CTaskBtnGroup__DrawBar_orig;
+void __cdecl CTaskBtnGroup__DrawBar_hook(
+    void *pThis,
+    HDC   hDC,
+    void *pRenderInfo,
+    PBUTTONRENDERINFOSTATES pRenderStates
+)
+{
+    LPRECT lprcDest = (LPRECT)((char *)pRenderInfo + 4);
+
+    HTHEME hTheme = OpenThemeData(NULL, L"BUTTON");
+
+    if (hTheme) {
+        int iStateId = PBS_NORMAL; 
+
+        if (pRenderStates->data[2]) {
+
+            iStateId = PBS_PRESSED; // or use PBS_HOT if you prefer
+
+        } else if (pRenderStates->data[4]) {
+
+            iStateId = PBS_PRESSED;
+
+        }
+
+        // Draw the button with the current theme.
+
+        DrawThemeBackground(hTheme, hDC, BP_PUSHBUTTON, iStateId, lprcDest, NULL);
+
+
+        // Close the theme data handle when done.
+
+        CloseThemeData(hTheme);
+    } else
+    {
+        // Fallback to non-themed drawing if themes are not available.
+
+        UINT uState = DFCS_BUTTONPUSH;
+        if (pRenderStates->data[2])
+        {
+            uState |= DFCS_CHECKED;
+        }
+        else if (pRenderStates->data[4])
+        {
+            uState |= DFCS_PUSHED;
+        }
+
+        DrawFrameControl(hDC, lprcDest, DFC_BUTTON, uState);
+    };
+
+
+    /* If button is pushed in, offset the rect for the icon and text draw */
+    if (pRenderStates->data[2]
+    ||  pRenderStates->data[4])
+    {
+        lprcDest->top++;
+        lprcDest->bottom++;
+        lprcDest->left++;
+        lprcDest->right++;
+    }
+
+    return;
+}
+
+BOOL Wh_ModInit(void)
+{
+    HMODULE hExplorer = GetModuleHandleW(NULL);
+
+    WindhawkUtils::SYMBOL_HOOK hooks[] = {
+        {
+            {
+                L"void __cdecl GDIHelpers::FillRectARGB(struct HDC__ *,struct tagRECT const *,unsigned char,unsigned long,bool)"
+            },
+            &GDIHelpers_FillRectARGB_orig,
+            GDIHelpers_FillRectARGB_hook,
+            FALSE
+        },
+        {
+            {
+                L"private: void __cdecl CTaskBtnGroup::_DrawBar(struct HDC__ *,struct BUTTONRENDERINFO const &,struct BUTTONRENDERINFOSTATES const &)"
+            },
+            (void **)&CTaskBtnGroup__DrawBar_orig,
+            (void *)CTaskBtnGroup__DrawBar_hook,
```

Comment:
Any reason you left the casts here?

Reply:
If I remove them the compilation fails.

Reply:
That's because the signature doesn't match, it should be:
```cpp
typedef void (*__cdecl CTaskBtnGroup__DrawBar_t)(void *, HDC, void *, PBUTTONRENDERINFOSTATES);
```
instead of:
```cpp
typedef void (*__cdecl CTaskBtnGroup__DrawBar_t)(void *, HDC, void *, void *);
```

Reply:
It works, thank you.